backend/keepalived/keepalived.go

```diff
@@ -328,23 +329,29 @@ func (k *Keepalived) alive(key string, prev liveness.State, leader bool) bool {
 func (k *Keepalived) dead(key string, prev liveness.State, leader bool) bool {
 	lager := logger.WithFields(logrus.Fields{
 		"status":          liveness.Dead.String(),
-		"previous_status": prev.String()})
+		"previous_status": prev.String(),
+		"is_leader":       fmt.Sprintf("%v", leader),
+	})
+
+	if !leader {
+		// If this client isn't the one that flipped the keepalive switch,
+		// don't do anything further.
+		logger.Debug("not the leader of this keepalive switch, stopping here")
+		return false
+	}
 
 	namespace, name, err := parseKey(key)
 	if err != nil {
+		// We couldn't parse the key, which probably means the key didn't contain a
+		// namespace. Log the error and then try to bury the key so it stops sending
+		// events to the watcher.
 		lager.Error(err)
-		return false
+		return true
 	}
 
 	lager = lager.WithFields(logrus.Fields{"entity": name, "namespace": namespace})
 	lager.Warn("keepalive timed out")
 
-	if !leader {
```

Comment:
Why has this been removed? This prevents the non-leader nodes from making unnecessary requests.

Reply:
The diff is confusing but it has only been moved up, on line 336, so we don't return `true`, and therefore bury the key if it's invalid and we are not the leader. Makes sense?

Reply:
I see! That makes sense, but wondering if it should happen after the logger gets its additional context (namespace, etc)

Reply:
Ah yes, I can move things around so we add the entity name & namespace before logging everything and then check the error.